src/System.Private.CoreLib/shared/System/DateTime.cs

```diff
@@ -1182,11 +1188,12 @@ public int Year
         // Checks whether a given year is a leap year. This method returns true if
         // year is a leap year, or false if not.
         //
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static bool IsLeapYear(int year)
         {
             if (year < 1 || year > 9999)
             {
-                throw new ArgumentOutOfRangeException(nameof(year), SR.ArgumentOutOfRange_Year);
+                ThrowHelper.ThrowArgumentOutOfRange_Year();
```

Comment:
is this leap year test faster - avoids modulo: `((year & 3) == 0 && ((year % 25) != 0 || (year & 15) == 0))`

https://stackoverflow.com/questions/3220163/how-to-find-leap-year-programatically-in-c/11595914#11595914

Reply:
This should work fine.

Reply:
although I'd reorder to put the bitwise before the modulo?
`((year & 3) == 0 && ((year & 15) == 0) || (year % 25) != 0)`

Reply:
Needs benchmarks. Against:
```csharp
return (uint)year % 4 == 0 && ((uint)year % 100 != 0 || (uint)year % 400 == 0); 
// (uint) cast helps JIT
```